src/Arcus.Security.Tests.Unit/Core/CachedSecretProviderTests.cs

```diff
@@ -119,7 +141,34 @@ public async Task CachedSecretProvider_TwoCallsOutsideCacheInterval_ShouldGetDif
         }
 
         [Fact]
-        public async Task CachedSecretProvider_SkipCached_ShouldGetDifferentValue()
+        public async Task CachedSecretProvider_GetSecret_TwoCallsOutsideCacheInterval_ShouldGetDifferentValue()
+        {
+            // Arrange
+            string secretKeyValue = Guid.NewGuid().ToString("N");
+            var testSecretProvider = new TestSecretProviderStub(secretKeyValue);
+            var memCache = new MemoryCache(new MemoryCacheOptions());
+            var cacheConfiguration = new CacheConfiguration(TimeSpan.FromMilliseconds(100));
+            string keyName = "MyValue";
+
+            // Act 
+            ICachedSecretProvider cachedSecretProvider = new CachedSecretProvider(testSecretProvider, cacheConfiguration, memCache);
+            Secret firstSecret = await cachedSecretProvider.GetSecret(keyName);
+            await Task.Delay(TimeSpan.FromMilliseconds(150));
+            string newSecretValue = Guid.NewGuid().ToString("N");
+            testSecretProvider.SecretValue = newSecretValue; // Change actual value on the internal secret provider !
+            Secret secondSecret = await cachedSecretProvider.GetSecret(keyName);
+
+            // Assert
+            Assert.True(firstSecret != null, "firstSecret != null");
```

Comment:
Use Assert.NotNull or how is it called?

Reply:
I suggest not to; because then you wouldn't know which once failled. Better for _Defect Localization_.

Reply:
Hm I don't think I'm following, can you elaborate on `then you wouldn't know which once failled` please?

Are you referring to `firstSecret` vs `secondSecret`? If so, the description of the assert should be able to help here.

Reply:
Yes, but only `Assert.True/False` allows a custom `userMessage` to be passed. :(

Reply:
Oh I love NUnit more than xUnit, this sucks. Ok then :(